### PR TITLE
MAUI-871321-UG documentation added for .NET MAUI SfStepProgressBar control

### PR DESCRIPTION
**Pull request by Priyadharshini-Periyasamy**
### **Description**

- Added UG content for the newly releasing SfStepProgressBar control in NET MAUI.

#### **Added topics**

- Overview

- Getting started

- Migration

### CODE REVIEW

File: MAUI/StepProgressBar/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+The Step progressbar control allows you to bind item collection by setting the [ItemsSource]() property from the [SfStepProgressBar](). Bind item collection in both XAML and C#.
+
+{% tabs %}
+{% highlight xaml tabtitle="XAML" hl_lines="8" %}
+
+<stepProgressBar:SfStepProgressBar
+                    x:Name="stepProgressBar"
+                    Margin="24,10,0,0"
+                    ActiveStepIndex="1"
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/getting-started.md:138]: remove margin

File: MAUI/StepProgressBar/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    <local:ViewModel />
+</ContentPage.BindingContext>
+
+{% endhighlight %}
+{% highlight c# tabtitle="C#" hl_lines="9" %}
+
+ViewModel viewModel = new ViewModel();
+SfStepProgressBar stepProgressBar = new SfStepProgressBar()
+{
+    Margin = new Thickness(24, 10, 0, 0),
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/getting-started.md:156]: remove

File: MAUI/StepProgressBar/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    Orientation = StepProgressBarOrientation.Horizontal,
+    ItemsSource = viewModel.StepProgressItem,
+};
+
+this.Content = stepProgressBar;
+
+{% endhighlight %}
+{% endtabs %}
+
+## Active step index value and Active progress value
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/getting-started.md:169]: Give proper title

File: MAUI/StepProgressBar/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+The [ActiveStepIndex]() property is used to represent index of the currently active step within the sequence of steps. The [ActiveStepProgressValue]() property is used to add the progress value of the currently active step within a sequence.
+
+{% tabs %}
+{% highlight xaml tabtitle="XAML" hl_lines="8" %}
+
+<stepProgressBar:SfStepProgressBar
+                    x:Name="stepProgressBar"
+                    Margin="24,10,0,0"
+                    ActiveStepIndex="2"
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/getting-started.md:178]: Remove margin if you used

File: MAUI/StepProgressBar/overview.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+documentation: ug
+---
+
+# Overview of .NET MAUI Step ProgressBar (SfStepProgressBar)
+
+Syncfusion .NET MAUI Step ProgressBar (SfStepProgressBar) control allows you to displays progress through multiple steps in a process, such as order tracking or a user registration form. It's customizable, with options for step shape, content, progress bar color, and more.
+
+## Key features
+
+* **Orientation options**: The Step ProgressBar allows you to change the orientation to horizontal or vertical direction.
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/overview.md:16]: options - needed?

Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/overview.md:16]: horizontal or vertical- Highlight keywords

File: MAUI/StepProgressBar/migration.md [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+<th>.NET MAUI SfStepProgressBar</th>
+<th>Description</th>
+</tr>
+<tr>
+<td>{{'[CompletedStepStyle]()'| markdownify }}</td>
+<td>{{'[CompletedStepSettings]()'| markdownify }}</td>
+<td>Gets or sets the completed step style for the step progressbar.</td>
+</tr>
+<tr>
+<td>{{'[InProgressStepStyle]()'| markdownify }}</td>
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/migration.md:201]: Screenshot 2024-03-12 at 6 18 42 PM
Description mentioned as style

File: MAUI/StepProgressBar/migration.md [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+<td>{{'[InProgressStepSettings]()'| markdownify }}</td>
+<td>Gets or sets the in progress step style for the step progressbar.</td>
+</tr>
+<tr>
+<td>{{'[NotStartedStepStyle]()'| markdownify }}</td>
+<td>{{'[NotStartedStepSettings]()'| markdownify }}</td>
+<td>Gets or sets the not started step style for the step progressbar.</td>
+</tr>
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/migration.md:209]: Gets or sets the label position or tile alignment.

Add description about maui property and its use case

File: MAUI/StepProgressBar/migration.md [hunk 8]
```diff
@@ -0,0 +1,425 @@
...
+<tr>
+<td>{{'[TitleSpace]()'| markdownify }}</td>
+<td>{{'[LabelSpacing]()'| markdownify }}</td>
+<td>Gets or sets the label spacing.</td>
+</tr>
+<tr>
+<td>{{'[LayoutOption]()'| markdownify }}</td>
+<td>Nil</td>
+<td>Not Supported.</td>
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/migration.md:233]: {{'[LayoutOption]()'\| markdownify }} | Nil | Not Supported.
-- | -- | --

Checked alternative solution available?"

File: MAUI/StepProgressBar/migration.md [hunk 9]
```diff
@@ -0,0 +1,425 @@
...
+<td>Gets or sets the size of the step.</td>
+</tr>
+<tr>
+<td>{{'[MarkerStrokeWidth]()'| markdownify }}</td>
+<td>{{'[StepStrokeWidth]()'| markdownify }}</td>
+<td>Gets or sets the stroke width of the step.</td>
+</tr>
+</table>
+
+### StepSettings
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/migration.md:247]: Screenshot 2024-03-12 at 6 23 26 PM

dont mention as text style- explain it is used for Primary text

File: MAUI/StepProgressBar/migration.md [hunk 10]
```diff
@@ -0,0 +1,425 @@
...
+<th>.NET MAUI SfStepProgressBar</th>
+<th>Description</th>
+</tr>
+<tr>
+<td>{{'[FontAttributes]()'| markdownify }}</td>
+<td>{{'[[FontAttributes]()(From [TextStyle]() property of [StepSettings]() class)]()'| markdownify }}</td>
+<td>Gets or sets the font attributes of the text style.</td>
+</tr>
+<tr>
+<td>{{'[FontColor]()'| markdownify }}</td>
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/migration.md:261]: ShapeType 

Description missed

File: MAUI/StepProgressBar/getting-started.md (hunk 1 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/getting-started.md:138]: Removed margin in all the files

File: MAUI/StepProgressBar/getting-started.md (hunk 2 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/getting-started.md:156]: Removed margin

File: MAUI/StepProgressBar/getting-started.md (hunk 3 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/getting-started.md:169]: I have changed the title as **Active step progression**

File: MAUI/StepProgressBar/getting-started.md (hunk 4 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/getting-started.md:178]: Removed margin

File: MAUI/StepProgressBar/migration.md (hunk 6 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/migration.md:201]: Step status descriptions are changed to Completed step settings

File: MAUI/StepProgressBar/migration.md (hunk 7 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/migration.md:209]: Added description about the maui API

File: MAUI/StepProgressBar/migration.md (hunk 9 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/migration.md:247]: I have changed the description as below
Gets or sets the font attributes of the text style which is used to customize the primary and secondary text.

File: MAUI/StepProgressBar/migration.md (hunk 10 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/migration.md:261]: Added description for the ShapeType.

File: MAUI/StepProgressBar/overview.md (hunk 5 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/overview.md:16]: Changed to Orientation

Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/overview.md:16]: Highlighted the keywords

File: MAUI/StepProgressBar/migration.md (hunk 8 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/migration.md:233]: Checked the LayoutOption API and we dont have alternative API in MAUI

File: MAUI/StepProgressBar/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,208 @@
+---
+layout: post
+title: Getting started with .NET MAUI Step ProgressBar control | Syncfusion
```
Comment (M-A32) [inline MAUI/StepProgressBar/getting-started.md:3]: title: Getting Started with .NET MAUI Step ProgressBar Control | Syncfusion

File: MAUI/StepProgressBar/getting-started.md [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+        }
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Populating step progressbar items
+
+You can use ItemsSource property of SfStepProgressBar to populate the step progressbar items. Hence, you must create a item collection and bind it to the progressbar control.
```
Comment (M-A32) [inline MAUI/StepProgressBar/getting-started.md:87]: You can use the ItemsSource property of SfStepProgressBar to populate the step progressbar items. Hence, you must create an item collection and bind it to the progressbar control.

File: MAUI/StepProgressBar/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+
+* Create a simple Observable Collection of item for the step progressbar,
```
Comment (M-A32) [inline MAUI/StepProgressBar/getting-started.md:89]: * Create a simple Observable Collection of the item for the step progressbar,

File: MAUI/StepProgressBar/migration.md [hunk 14]
```diff
@@ -0,0 +1,425 @@
...
+title: Migrate from Syncfusion Xamarin SfStepProgressBar to MAUI SfStepProgressBar
+description: Learn here all about Migrating from Syncfusion Xamarin step progressbar to Syncfusion .NET MAUI SfStepProgressBar control.
+platform: maui
+control: SfStepProgressBar
+documentation: ug
+---  
+
+# Migrate from Xamarin.Forms SfStepProgressBar to MAUI SfStepProgressBar
+
+To make the migration from the [Xamarin SfStepProgressBar](https://www.syncfusion.com/xamarin-ui-controls/xamarin-stepprogressbar) to [.NET MAUI SfStepProgressBar]() easier, most of the APIs from the Xamarin SfStepProgressBar were kept in the.NET MAUI SfStepProgressBar. However, to maintain the consistency of API naming in the .NET MAUI SfStepProgressBar, some of the APIs have been renamed. Please find the difference in the following topics.
```
Comment (M-A32) [inline MAUI/StepProgressBar/migration.md:12]: To make the migration from the [Xamarin SfStepProgressBar](https://www.syncfusion.com/xamarin-ui-controls/xamarin-stepprogressbar) to [.NET MAUI SfStepProgressBar]() easier, most of the APIs from the Xamarin SfStepProgressBar were kept in the.NET MAUI SfStepProgressBar. However, to maintain the consistency of API naming in the .NET MAUI SfStepProgressBar, some of the APIs have been renamed. Find the difference in the following topics.

File: MAUI/StepProgressBar/overview.md [hunk 15]
```diff
@@ -0,0 +1,36 @@
+---
+layout: post
+title: About .NET MAUI Step ProgressBar control | Syncfusion
```
Comment (M-A32) [inline MAUI/StepProgressBar/overview.md:3]: title: About .NET MAUI Step ProgressBar Control | Syncfusion

**Review by M-A32 — CHANGES_REQUESTED** [review]
I have completed the review.

File: MAUI/StepProgressBar/overview.md (hunk 15 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/overview.md:3]: Content changed

File: MAUI/StepProgressBar/migration.md (hunk 14 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/migration.md:12]: Content changed

File: MAUI/StepProgressBar/getting-started.md (hunk 13 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/getting-started.md:89]: Content changed

File: MAUI/StepProgressBar/getting-started.md (hunk 12 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/getting-started.md:87]: Content changed

File: MAUI/StepProgressBar/getting-started.md (hunk 11 above)
Comment (Priyadharshini-Periyasamy) [reply MAUI/StepProgressBar/getting-started.md:3]: Content changed

File: MAUI/StepProgressBar/migration.md [hunk 16]
```diff
@@ -0,0 +1,425 @@
...
+<th>Xamarin SfStepProgressBar</th>
+<th>.NET MAUI SfStepProgressBar</th>
+<th>Description</th>
+</tr>
+<tr>
+<td>{{'[StepTapped]()'| markdownify }}</td>
+<td>{{'[StepTapped]()'| markdownify }}</td>
+<td>Occurs whenever the progressbar step is tapped.</td>
+</tr>
+<tr>
```
Comment (MuniappanSubramanian) [inline MAUI/StepProgressBar/migration.md:420]: Add upcoming featues

**Comment by MuniappanSubramanian** [comment]
rebuild

**Review by MuniappanSubramanian — APPROVED** [review]

**Pull request merged**